File: grokking_algorithms/05_hash_tables/golang/02_check_voter.go
/*
* @Author: Clarence
* @Date:   2020-01-19 11:46:52
* @Last Modified by:   Clarence
* @Last Modified time: 2020-01-19 11:51:04
*/
package main 

import "fmt"

var voted map[string]bool 

func main()  {
	voted = make(map[string]bool)
	check_voter("tom")
	check_voter("mike")
	check_voter("mike")
}

func check_voter(name string) {
	if voted[name] {
		fmt.Println("kick them out!")
	} else {
		voted[name] = true 
		fmt.Println("let them vote!")
	}
}

File: grokking_algorithms/05_hash_tables/python/02_check_voter.py
# -*- coding: utf-8 -*-
# @Author: Clarence
# @Date:   2020-01-19 11:41:26
# @Last Modified by:   Clarence
# @Last Modified time: 2020-01-19 11:44:26
voted = {} 
def check_voter(name):
	if voted.get(name):
		print("kick them out!")
	else:
		voted[name] = True 
		print("let them vote!")

check_voter("tom")
check_voter("mike")
check_voter("mike")
